File: Go-Solutions/nowcoder/jz14.go
package nowcoder

func FindKthToTailjz14(pListHead *ListNode, k int) *ListNode {
	// write code here
	if pListHead == nil {
		return nil
	}

	length := getLenjz14(pListHead)

	if k > length {
		return nil
	}

	node := pListHead
	for i := 0; i < length-k; i++ {
		node = node.Next
	}
	return node
}

func getLenjz14(listHead *ListNode) int {
	count := 0
	head := listHead
	for head != nil {
		count++
		head = head.Next
	}
	return count
}

File: Java-Kotlin/src/main/java/jianzhi/jz14.java
package jianzhi;

import common.datastructure.ListNode;

public class jz14 {

    public ListNode FindKthToTail(ListNode head, int k) {
        if (head == null) {
            return null;
        }

        int length = getLen(head);
        if (k > length) {
            return null;
        }

        ListNode node = head;
        for (int i = 0; i < length - k; i++) {
            node = node.next;
        }
        return node;
    }

    private int getLen(ListNode head) {
        ListNode runner = head;
        int count = 0;
        while (runner != null) {
            count++;
            runner = runner.next;
        }

        return count;
    }
}
